src/main/kotlin/app/extractors/ExtractorInterface.kt

```diff
@@ -54,22 +56,70 @@ interface ExtractorInterface {
             return pmml
         }
 
+        private fun downloadModel(name: String, outputDir: String) {
+            val url = BuildConfig.LIBRARY_MODELS_URL + "$name.pkl.z"
+            val outputPath = "$outputDir/$name.pkl.z"
+
+            if (Files.notExists(Paths.get(outputDir))) {
+                Files.createDirectories(Paths.get(outputDir))
+            }
+
+            val builder = HttpClientBuilder.create()
+            val client = builder.build()
+            try {
+                client.execute(HttpGet(url)).use { response ->
+                    val entity = response.entity
+                    if (entity != null) {
+                        FileOutputStream(outputPath).use { outstream ->
+                            entity.writeTo(outstream)
+                            outstream.flush()
+                            outstream.close()
+                        }
+                    }
+
+                }
+            }
+            catch (e: Exception) {
+                Logger.error(e, "Failed to download $name model")
+            }
+        }
+
+        private fun getPickleInputStream(pklPath: String): FileInputStream {
+            val pklStream = try {
+                FileInputStream(pklPath)
+            }
+            catch (e: Exception) {
+                Logger.error(e, "Failed to load $pklPath")
+                FileInputStream(pklPath)
```

Comment:
Actually, in case of problems with loading pickle files exception should be throw and handled in top classes, cuz in that case we can't extract data at all. Second call may help in some cases but not in all. Remove it.